src/Approximations/overapproximate.jl

```diff
@@ -546,23 +546,50 @@ Overapproximating a set with template directions.
 ### Output
 
 A polyhedron overapproximating the set `X` with the directions from `dir`.
-If the directions are known to be bounded, the result is an `HPolytope`,
-otherwise the result is an `HPolyhedron`.
+The overapproximation is computed using support functions. If the obtained set is
+bounded, the result is an `HPolytope`. Otherwise the result is an `HPolyhedron`.
 """
 function overapproximate(X::LazySet{N}, dir::AbstractDirections{N, VN}) where {N, VN}
-    halfspaces = Vector{LinearConstraint{N, VN}}()
-    sizehint!(halfspaces, length(dir))
-    T = isbounding(dir) ? HPolytope : HPolyhedron
-    H = T(halfspaces)
+    H = Vector{LinearConstraint{N, VN}}()
+    sizehint!(H, length(dir))
+
     for d in dir
-        addconstraint!(H, LinearConstraint(d, ρ(d, X)))
+        sf = ρ(d, X)
+        if !isinf(sf)
+            addconstraint!(H, LinearConstraint(d, sf))
+        end
+    end
+    remove_redundant_constraints!(H) || throw(ArgumentError("unable to remove redundant constraints"))
+    P = HPolyhedron(H)
```

Comment:
i didn't see a good reason to use `isbounding(dir)`, because what matters is the set after the overapproximation. (it can be bounded or not for cases in which dir is bounded or not).

Reply:
If the input is bounded and the directions are bounding, then the result is bounded. I agree that we cannot conclude in all other cases, but this should be the most common case I guess. As you prefer.

Reply:
ok, i see. we can do:

```julia
if isbounded(X) && isbounding(dirs)
    return HPolytope(H)
end
P = HPolyhedron(...)
```
i dont know if this is better or worse than the current code

Reply:
Neither do I. In some cases this information is constant-time while I guess `_isbounded_stiemke` is not.

Reply:
i made [that change](https://github.com/JuliaReach/LazySets.jl/pull/2405/files#r525524957) in the latest commit